sentry/src/main/java/io/sentry/Sentry.java

```diff
@@ -1,37 +1,33 @@
 package io.sentry;
 
-import io.sentry.config.ResourceLoader;
+import io.sentry.config.Lookup;
 import io.sentry.context.Context;
 import io.sentry.dsn.Dsn;
 import io.sentry.event.Breadcrumb;
 import io.sentry.event.Event;
 import io.sentry.event.EventBuilder;
 import io.sentry.event.User;
+import io.sentry.util.Nullable;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.concurrent.atomic.AtomicBoolean;
-
 /**
  * Sentry provides easy access to a statically stored {@link SentryClient} instance.
  */
 public final class Sentry {
     private static final Logger logger = LoggerFactory.getLogger(Sentry.class);
+
     /**
-     * The most recently constructed {@link SentryClient} instance, used by static helper
-     * methods like {@link Sentry#capture(Event)}.
-     */
-    private static volatile SentryClient storedClient = null;
-    /**
-     * Tracks whether the {@link #init()} method has already been attempted automatically
-     * by {@link #getStoredClient()}.
+     * A synchronization guard of the stored client. Not using the class as the guard because that could theoretically
+     * deadlock with 3rd party code if it also synced on the class.
      */
-    private static AtomicBoolean autoInitAttempted = new AtomicBoolean(false);
+    private static final Object STORED_CLIENT_ACCESS = new Object();
```

Comment:
Is this change needed to get rid of testNG?
Could you please elaborate why move from an atomic operation to a lock?